libs/execution_context/ebpf_native.c

```diff
@@ -2356,15 +2407,97 @@ _ebpf_native_helper_address_changed(
 }
 
 _Must_inspect_result_ ebpf_result_t
-ebpf_native_authorize_module(_In_ const uint8_t* module_hash)
+ebpf_native_authorize_module(_In_ const GUID* module_id, _In_ const uint8_t* module_hash)
 {
     EBPF_LOG_ENTRY();
-    uint32_t authorized = 1;
     ebpf_result_t result = EBPF_SUCCESS;
 
+    ebpf_native_authorized_module_entry_t entry = {0};
+
+    memcpy(entry.expected_module_hash, module_hash, sizeof(entry.expected_module_hash));
+    entry.insertion_time = cxplat_query_time_since_boot_approximate(false) * EBPF_NS_PER_FILETIME;
+
     ebpf_lock_state_t state = ebpf_lock_lock(&_ebpf_native_authorized_module_table_lock);
     result = ebpf_hash_table_update(
```

Comment:
question: what if the same key / value already exists? should we absorb the error? It is possible user app called `verify_and_authorize_native_image()` and then crashed before actually loading the native module. In that case, it can attempt to re-authorize the same native module.